### PR TITLE
CSHARP-2285

### PR DESCRIPTION
**Pull request by vincentkam**
Update Feature class to include when features stop being supported.
Disable group command tests.

### CODE REVIEW

File: tests/MongoDB.Driver.Core.Tests/Core/Operations/GroupOperationTests.cs [hunk 1]
```diff
@@ -405,7 +405,7 @@ public void CreateCommand_should_return_expected_result_when_MaxTime_is_set(long
             [Values(false, true)]
             bool async)
         {
-            RequireServer.Check().Supports(Feature.Collation);
+            RequireServer.Check().Supports(Feature.GroupCommand).Supports(Feature.Collation);
```
Comment (rstam) [inline tests/MongoDB.Driver.Core.Tests/Core/Operations/GroupOperationTests.cs]: Supports accepts multiple features. You can use:

    ...Suports(Feature.GroupCommand, Feature.Collation);

File: tests/MongoDB.Driver.Core.Tests/Core/Operations/GroupOperationTests.cs [hunk 2]
```diff
@@ -561,7 +561,7 @@ public void CreateCommand_should_return_expected_result_when_MaxTime_is_set(long
         public void Execute_should_throw_when_maxTime_is_exceeded(
             [Values(false, true)] bool async)
         {
-            RequireServer.Check().Supports(Feature.FailPoints).ClusterTypes(ClusterType.Standalone, ClusterType.ReplicaSet);
+            RequireServer.Check().Supports(Feature.GroupCommand).Supports(Feature.FailPoints).ClusterTypes(ClusterType.Standalone, ClusterType.ReplicaSet);
```
Comment (rstam) [inline tests/MongoDB.Driver.Core.Tests/Core/Operations/GroupOperationTests.cs]: You can combine the two calls to Support.

**Review by rstam — COMMENTED** [review]
Check Evergreen.

There are a few more tests that still need to be skipped.

**Comment by vincentkam** [comment]
Disabled some additional tests, waiting on Evergreen.

File: tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs [hunk 3]
```diff
@@ -2048,6 +2048,7 @@ public void TestGetMore()
         [Fact]
```
Comment (rstam) [inline tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs]: Needs to be [SkippableFact]

File: tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs [hunk 4]
```diff
@@ -2076,6 +2077,7 @@ public void TestGroupWithFinalizeFunction()
...
```
Comment (rstam) [inline tests/MongoDB.Driver.Legacy.Tests/MongoCollectionTests.cs]: Needs to be [SkippableFact]

Look for all of them. I won't add a comment to each one...

**Review by rstam — CHANGES_REQUESTED** [review]

**Comment by rstam** [comment]
Notify me once Evergreen is green (or only failing on unrelated tests) and I'll lgtm this.

**Comment by vincentkam** [comment]
Green except for the expected errors: https://evergreen.mongodb.com/version/5b118d53e3c3314e54f0479d

**Review by rstam — APPROVED** [review]

**Pull request merged**